src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.BlazorWasmCompanionExtension/src/extension.ts

```diff
@@ -0,0 +1,80 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { spawn } from 'child_process';
+import { acquireDotnetInstall } from './acquireDotnetInstall';
+import { getAvailablePort } from "./getAvailablePort";
+
+let launchDebugProxy: vscode.Disposable;
+let killDebugProxy: vscode.Disposable;
+
+export function activate(context: vscode.ExtensionContext) {
+    const outputChannel = vscode.window.createOutputChannel("Blazor WASM Debug Proxy");
+    const pidsByUrl = new Map<string, number>();
+
+    launchDebugProxy = vscode.commands.registerCommand('ms-blazorwasm-companion.launchDebugProxy', async () => {
+        try {
+            const debuggingPort = await getAvailablePort(9222);
+            const debuggingHost = `http://localhost:${debuggingPort}`;
+
+            let dotnet = "dotnet";
+            // The vscode.env.remoteName property is set when connected to
+            // any kind of remote workspace. See
+            // https://code.visualstudio.com/api/advanced-topics/remote-extensions#varying-behaviors-when-running-remotely-or-in-the-codespaces-browser-editor
+            const isRemote = vscode.env.remoteName !== 'undefined';
+            if (isRemote) {
+                dotnet = await acquireDotnetInstall(outputChannel);
+            }
+
+            const debugProxyLocalPath = `${context.extensionPath}/BlazorDebugProxy/BrowserDebugHost.dll`;
+            outputChannel.appendLine(`Launching debugging proxy from ${debugProxyLocalPath}`);
+            const spawnedProxy = spawn(dotnet,
+                [debugProxyLocalPath , '--DevToolsUrl', debuggingHost],
+                { detached: process.platform !== 'win32' });
```

Comment:
Can you add some clarity here? Makes me think if it's an x86 dotnet then the companion doesn't do anything?

Reply:
Running the process `detached` allows it to run in the background. The behavior between Windows/Unix is a little different as documented [here](https://nodejs.org/api/child_process.html#child_process_options_detached). Specifically, Windows will attempt to launch in a separate console window.

However, after some contemplation, I think it's sensible to keep the process attached to the parent (VS Code in this case). Provides another layer of security that it will be killed when VS Code closes.